src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/Runtime/ThreadStatics.cs

```diff
@@ -22,6 +22,34 @@ internal static class ThreadStatics
         /// static storage for the given type.
         /// </summary>
         internal static unsafe object GetThreadStaticBaseForType(TypeManagerSlot* pModuleData, int typeTlsIndex)
+        {
+            if (typeTlsIndex >= 0)
+                return GetUninlinedThreadStaticBaseForType(pModuleData, typeTlsIndex);
+
+            ref object? threadStorage = ref RuntimeImports.RhGetInlinedThreadStaticStorage();
+            if (threadStorage != null)
+                return threadStorage;
+
+            return GetInlinedThreadStaticBaseSlow(ref threadStorage);
+        }
+
+        [RuntimeExport("RhpGetInlinedThreadStaticBaseSlow")]
+        internal static unsafe object GetInlinedThreadStaticBaseSlow(ref object? threadStorage)
+        {
+            Debug.Assert(threadStorage == null);
+            // Allocate an object that will represent a memory block for all thread static fields
+            TypeManagerHandle typeManager = RuntimeImports.RhGetSingleTypeManager();
```

Comment:
This is the only place where we are using `RhGetSingleTypeManager`. Can we pass it in through `pModuleData` argu,ent of `GetThreadStaticBaseForType` instead? It will move us closer to a scheme where the thread-static is emitted by the compiler.

Reply:
I tried to avoid passing arguments to `RhpGetInlinedThreadStaticBase` since it makes the code simpler on Unix. And the extra arguments would be used only once per thread's life time while we would always have to pass the extra arguments.  
Ideally the slow path should be able to figure the typemanager by itself. It would be even ok to have it somewhat expensive, since it will be called very rarely. 

The only information that slow path gets right now is the location of the root. But there could be more information at the location. 
Maybe the location could be something like `{rootRef{0}, pTypeManager{relocated}, Next{0}}`, so that the slow path could fetch the Type Manager from the location. The scenario can't be simulated easily though, when the root is in the native runtime.

Reply:
I will reintroduce the type manager parameter just to wrap this up, but we should think if there is a better way.

Reply:
Once this is inlined by the JIT, the type manager argument should be only on the slow path - it should not be materialized in the inlined fast path.

I agree we may consider packing all thread static information into a descriptor and let this descriptor to be the only argument that gets passed into the slow path. I would worry about it once we get the JIT inlining in place. I will be easier to reason about how to optimize the inlined fast path for speed and the slow path for size.

Reply:
(If you think that it is not worth it to clean this up, I am ok with merging this PR as is.)

Reply:
Ok, makes sense. I'll spend some more time on figuring what is going on in INLINE_GET_TLS_VAR` and how that can be simplified. I'd need to have a good understanding of that pattern anyways and why/whether it needs to differ from x64. 

I will leave `RhGetSingleTypeManager` in place for now. It will likely be not needed at next steps, but what happens with asm helpers is less clear.